File: code/ch19/19.1.2. 189.rotate.cpp
#include <vector>
using namespace std;

class Solution {
public:
  void rotate(vector<int>& nums, int k) {
    vector<int> copy = nums;
    int n = nums.size();
    if (k % n == 0) return;
    for (int i = 0; i < n; i++)
      nums[(k + i) % n] = copy[i];
  }
};

File: code/ch19/19.1.2. 189.rotate.java
public class Solution {
  public void rotate(int[] nums, int k) {
    int[] copy = new int[nums.length];
    int n = nums.length;
    for (int i = 0; i < n; i++) copy[i] = nums[i];
    if (k % n == 0) return;
    for (int i = 0; i < n; i++)
      nums[(k + i) % n] = copy[i];
  }
}
